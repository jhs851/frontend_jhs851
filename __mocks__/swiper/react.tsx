import React from 'react'

const mock: React.FC = ({ children }) => <div>{children}</div>

export default mock
