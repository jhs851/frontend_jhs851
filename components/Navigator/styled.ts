import styled from 'styled-components'
import { flexCenter } from 'styles/flex'

export const StyledNav = styled.header`
  height: 44px;
  background-color: ${({ theme }) => theme.primary};
  position: relative;
  ${flexCenter}
`

export const HeaderLeftContainer = styled.div`
  position: absolute;
  left: 12px;
  top: 50%;
  transform: translateY(-50%);
`

export const HeaderRightContainer = styled.div`
  position: absolute;
  right: 12px;
  top: 50%;
  transform: translateY(-50%);
`
