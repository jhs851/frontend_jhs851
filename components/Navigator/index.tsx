import React, { ReactElement } from 'react'
import { HeaderRightContainer, HeaderLeftContainer, StyledNav } from './styled'

type P = {
  headerLeft?: ReactElement
  headerRight?: ReactElement
  title?: ReactElement
}

const Navigator: React.FC<P> = ({ headerLeft, headerRight, title }) => {
  return (
    <StyledNav>
      <HeaderLeftContainer>{headerLeft}</HeaderLeftContainer>
      {title}
      <HeaderRightContainer>{headerRight}</HeaderRightContainer>
    </StyledNav>
  )
}

export default Navigator
