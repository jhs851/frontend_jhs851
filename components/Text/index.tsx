import React from 'react'
import { StyledText } from './styled'
import { Colors } from 'types/theme'

type P = {
  size: number
  weight?: 500 | 600 | 700
  letterSpacing?: number
  color?: keyof Colors
  className?: string
  ellipsis?: boolean
}

const Text: React.FC<P> = ({
  size,
  weight,
  letterSpacing,
  color,
  ellipsis = false,
  className,
  children,
}) => (
  <StyledText
    size={size}
    weight={weight}
    letterSpacing={letterSpacing}
    color={color}
    ellipsis={ellipsis}
    className={className}
  >
    {children}
  </StyledText>
)

export default Text
