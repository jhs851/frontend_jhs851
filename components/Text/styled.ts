import styled, { css } from 'styled-components'
import { Colors } from 'types/theme'

export const StyledText = styled.p<{
  size: number
  weight?: 500 | 600 | 700
  letterSpacing?: number
  color?: keyof Colors
  ellipsis: boolean
}>`
  font-family: AppleSDGothicNeo, sans-serif;
  font-size: ${({ size }) => `${size}px`};
  font-weight: ${({ weight }) => weight || 500};
  letter-spacing: ${({ letterSpacing }) => `${letterSpacing || -0.1}`}px;
  color: ${({ theme, color }) => theme[color || 'foreground']};

  ${({ ellipsis }) =>
    ellipsis &&
    css`
      overflow: hidden;
      text-overflow: ellipsis '..';
      white-space: nowrap;
    `}
`
