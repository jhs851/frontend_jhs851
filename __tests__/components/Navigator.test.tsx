import { render } from '@testing-library/react'
import Navigator from 'components/Navigator'
import Text from 'components/Text'

describe('Navigator 컴포넌트', () => {
  it('렌더링이 올바르게 된다', () => {
    const { container } = render(
      <Navigator
        headerLeft={<Text size={14}>왼쪽</Text>}
        title={<Text size={14}>제목</Text>}
        headerRight={<Text size={14}>오른쪽</Text>}
      />,
    )

    expect(container).toMatchSnapshot()
  })
})
