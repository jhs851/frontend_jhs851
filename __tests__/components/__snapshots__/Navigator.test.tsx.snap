// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator 컴포넌트 렌더링이 올바르게 된다 1`] = `
.c0 {
  height: 44px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  position: absolute;
  left: 12px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c3 {
  position: absolute;
  right: 12px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c2 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 14px;
  font-weight: 500;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <p
        class="c2"
        size="14"
      >
        왼쪽
      </p>
    </div>
    <p
      class="c2"
      size="14"
    >
      제목
    </p>
    <div
      class="c3"
    >
      <p
        class="c2"
        size="14"
      >
        오른쪽
      </p>
    </div>
  </header>
</div>
`;
